### PR TITLE
Use a single handler for executing all commands

### LINKED ISSUE #6792: Many handlers are implemented for executing commands
/kind user-story

## User Story

Different parts of odo (odo dev build command, odo dev run/debug command, odo deploy, postStart, preStop events), are using different handlers for handling command execution.

Especially, "exec on container" commands are handled differently based on from where they call them.

It would be interesting to have the same handler, determining if the container already exists or not, and to execute the command accordingly
- on an existing container
- on a job

 
## Acceptance Criteria 
- [ ] It should ....
- [ ] It should ....

## Links 
- Related Epic (mandatory):

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this:**

/kind code-refactoring

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #6792 

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: pkg/component/handler_test.go [hunk 1]
```diff
@@ -0,0 +1,1764 @@
...
+				if (err != nil) != tt.wantErr {
+					t.Errorf("Err expected %v, got %v", tt.wantErr, err)
+				}
+			}
+		})
+	}
+}
+
+/*
+func TestApplyImage(t *testing.T) {
```
Comment (rm3l) [inline pkg/component/handler_test.go:1535]: Can we remove these commented lines if not needed? I can see they seem to be covered with the big test above..

File: pkg/component/exec_handler_test.go
Comment (rm3l) [inline pkg/component/exec_handler_test.go:1]: Do we still need this test file?

**Review by rm3l — COMMENTED** [review]
Thanks for your work on this! LGTM overall.
Had a few minor comments..

File: pkg/component/handler_test.go (hunk 1 above)
Comment (feloy) [reply pkg/component/handler_test.go]: I forgot to cleanup these old tests. Thanks

**Review by rm3l — APPROVED** [review]

**Comment by feloy** [comment]
/override 

```

**Comment by feloy** [comment]
/override OpenShift-Integration-tests/OpenShift-Integration-tests

```
  [oc] Error from server (InternalError): error when creating "/go/odo_1/tests/examples/manifests/bindablekind-instance.yaml": Internal error occurred: failed calling webhook "mcluster.kb.io": failed to call webhook: Post "https://postgresql-operator-controller-manager-1-18-3-service.openshift-operators.svc:443/mutate-postgresql-k8s-enterprisedb-io-v1-cluster?timeout=10s": context deadline exceeded
```

**Comment by feloy** [comment]
/override windows-integration-test/Windows-test
```
  [oc] ''Unable to connect to the server: dial tcp 161.156.183.218:30329: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.

```

**Pull request merged**